lib/appcanary/configuration.rb

```diff
@@ -31,27 +59,78 @@ def configure(&block)
       block.call(configuration) if block
     end
 
+    # another way to do static configuration
+    def api_key=(val);      configuration.api_key = val;      end
+    def gemfile_lock=(val); configuration.gemfile_lock = val; end
+    def monitor_name=(val); configuration.monitor_name = val; end
+    def base_uri=(val);     configuration.base_uri = val;     end
+
+    # throws if config is broken
+    def check_runtime_config!(endpoint, config)
+      if endpoint == :monitors && config[:monitor_name].nil?
+        raise Appcanary::ConfigurationError.new("Appcanary.monitor_name = ???")
+      end
+    end
+
     def resolved_config
```

Comment:
1. By rendering this into a hash that initializes the Client _once_, this means that if you screw up configuration you can't change it later, i.e.

```ruby
Appcanary.api_key = "foo"
Appcanary.vulnerable? # => error
Appcanary.api_key = "correct key"
Appcanary.vulnerable? # => same error
```

this would break my expectations.

Suggestion: just pass in the configuration object! It's messy-mutable-state but so long as _we_ don't mutate it ¯\\\_(ツ)_/¯  

`resolve_config` should check whether config is valid (api_key, gemfile_lock_path); if it's not, try loading in a yaml file then check if it's valid again; then throw corresponding errors

but then it should pass the configuration object down to the client!

Does that make sense?

2. If we're looking for a file, can we use [`Dir.pwd`](http://ruby-doc.org/core-2.4.0/Dir.html#method-c-pwd) instead of Bundle.root? I confess don't know how that works inside a gem but should do the trick. Not that I expect this to run outside of Bundler.